contracts/utils/EmergencyBrake.sol

```diff
@@ -1,191 +1,282 @@
+    /// @param user address with an isolation plan
+    function check(address user)
+        external view override returns (bool)
+    {
+        Plan storage plan_ = plans[user];
+
+        // Loop through the ids array, and check all roles.
+        uint length = uint(plan_.ids.length);
+        require(length > 0, "Plan not found");
+
+        for (uint i = 0; i < length; ++i ) {
+            bytes32 id = plan_.ids[i];
+            Permission memory permission_ = plan_.permissions[id]; 
+            AccessControl host = AccessControl(permission_.host);
+
+            if (!host.hasRole(permission_.signature, user)) return false;
+        }
+
+        return true;
     }
 
     /// @dev Execute an access removal transaction
-    function execute(address target)
+    /// @notice The plan needs to be kept up to date with the current permissioning, or it will revert.
+    /// @param user address with an isolation plan
+    function execute(address user)
         external override auth
     {
-        require(plans[target].state == State.PLANNED, "Emergency not planned for.");
-        plans[target].state = State.EXECUTED;
+        Plan storage plan_ = plans[user];
+        require(!plan_.executed, "Already executed");
+        plan_.executed = true;
 
-        Permission[] memory permissions_ = plans[target].permissions;
+        // Loop through the ids array, and revoke all roles.
+        uint length = uint(plan_.ids.length);
+        require(length > 0, "Plan not found");
 
-        for (uint256 i = 0; i < permissions_.length; i++){
-            // AccessControl.sol doesn't revert if revoking permissions that haven't been granted
-            // If we don't check, planner and executor can collude to gain access to contacts
-            Permission memory permission_ = permissions_[i]; 
-            AccessControl contact = AccessControl(permission_.contact);
+        for (uint i = 0; i < length; ++i ) {
+            bytes32 id = plan_.ids[i];
+            Permission memory permission_ = plan_.permissions[id]; 
+            AccessControl host = AccessControl(permission_.host);
+
+            // `revokeRole` won't revert if the role is not granted, but we need
+            // to revert because otherwise operators with `execute` and `restore`
+            // permissions will be able to restore removed roles if the plan is not
+            // updated to reflect the removed roles.
+            // By reverting, a plan that is not up to date will revert on execution,
+            // but that seems like a lesser evil versus allowing operators to override
+            // governance decisions.
             bytes4 signature_ = permission_.signature;
             require(
-                contact.hasRole(signature_, target),
+                host.hasRole(signature_, user),
                 "Permission not found"
             );
-            contact.revokeRole(signature_, target);
+            host.revokeRole(permission_.signature, user);
```

Comment:
@alcueca 
nit - is there a reason for creating the signature_ variable and then using it on the next line, but then not using it on the line after that?  seems more consistent either to use the variable in both places or just get rid of it altogether
```suggestion
            host.revokeRole(signature_, user);
```